### ISSUE TITLE
vmctl throws error when imported measurement is empty (from influxdb to vm)

### ISSUE THREAD

**Issue by madface303** [VictoriaMetrics/VictoriaMetrics#3837]
### Describe the bug

I try to import my influxdb database with vmctl with the following command:

`./vmctl-prod influx --influx-user homeassistant --influx-password mysupersecretpw --influx-database homeassistant`

I am getting the following error:

```
2023/01/01 15:03:52 Exploring scheme for database "homeassistant"
2023/01/01 15:03:52 fetching fields: command: "show field keys"; database: "homeassistant"; retention: "autogen"
2023/01/01 15:03:52 found 80 fields; skipped 99 non-numeric fields
2023/01/01 15:03:52 fetching series: command: "show series"; database: "homeassistant"; retention: "autogen"
2023/01/01 15:03:52 found 404 series
2023/01/01 15:03:52 explore query failed: can't find field keys for measurement "ETHUSD"
```

### To Reproduce

It happens every time vmctl-prod runs and a measurement is empty (in the retention policy trying to import)

### Version

I'm using home assistant add-on
VictoriaMetrics 1.86.1

vmctl-prod  1.87.1 (just tried again with the latest version)

### Additional information

I found the reason by myself, maybe it may help the others:

My setup in influxdb has 3 retention policies. Autogen (this is the one used by the import from vmctl above), a retention policy between 6 months to 2 years and a third one with data older than 2 years.
The above error occurs when a measurement has no data in the retention policy trying to import. When deleting the measurement (and all other that have no data in the actual retention policy), the import will work as expected.

**Comment by dmitryk-dk** [comment]
Hi @madface303 ! Can you share some examples of the data? It will helps to understand the problem and fix it

**Comment by madface303** [comment]
Not really shure what you really need but i will try:
For the above example the data would look like in influxdb:
```
> select * from "autogen"."ETHUSD"
> select * from "y2_5m"."ETHUSD"
name: ETHUSD
time                domain entity_id       value
----                ------ ---------       -----
1619279700000000000 sensor bitfetch_ethusd 2252.19
> select * from "inf_15m"."ETHUSD"
>
```
When i delete this measurement the next error would be: 
`explore query failed: can't find field keys for measurement "ETHEUR"`

For this measurement there is a little bit more data:
```
> select * from "autogen"."ETHEUR" limit 10
> select * from "y2_5m"."ETHEUR" limit 10
name: ETHEUR
time                domain entity_id       value
----                ------ ---------       -----
1619280000000000000 sensor bitfetch_etheur 1870.622
1619280300000000000 sensor bitfetch_etheur 1865.065
1619283900000000000 sensor bitfetch_etheur 1856.87
1619287500000000000 sensor bitfetch_etheur 1865.58
1619291100000000000 sensor bitfetch_etheur 1885.42
1619294700000000000 sensor bitfetch_etheur 1884.86
1619298300000000000 sensor bitfetch_etheur 1900.01
1619301900000000000 sensor bitfetch_etheur 1895.57
1619305500000000000 sensor bitfetch_etheur 1866.97
1619309100000000000 sensor bitfetch_etheur 1849.32
> select * from "inf_15m"."ETHEUR" limit 10
>
```

Both measurements have no data in the autogen policy and have throwed the error. Deleting each measurement which has no data in the autogen policy solved it for me, the import ran fine and had no more errors (just to mention, deleting a measurement also deletes data in all retention policies, so i had to restore backups several times to import my database and do the same thing for each retention policy).

**Comment by dmitryk-dk** [comment]
HI @madface303 ! Thank you for the explanation. We will take a look what we can do

**Comment by madface303** [comment]
Yes sure, i tried again with this flag with 2 retention policies (--influx-retention-policy autogen   and    --influx-retention-policy y2_5m , where y2_5m is the name of another retention policy), and the problem remains. As i read in the docs, if the flag is not set vmctl should use the autogen policy for default.

**Comment by madface303** [comment]
Here are the steps to reproduce the error easily:
```

influx -precision rfc3339

> CREATE DATABASE NOAA_water_database
> exit

curl https://s3.amazonaws.com/noaa.water-database/NOAA_data.txt -o NOAA_data.txt

influx -import -path=NOAA_data.txt -precision=s -database=NOAA_water_database

influx -precision rfc3339 -database NOAA_water_database

> CREATE RETENTION POLICY "y2_5m" ON "NOAA_water_database" DURATION 1800d REPLICATION 1
> select * into tempm from "h2o_feet"
> delete from "h2o_feet"
> select * into "y2_5m"."h2o_feet" from tempm

./vmctl-prod influx --influx-database NOAA_water_database --vm-addr http://10.10.11.88:8428 --influx-retention-policy autogen
InfluxDB import mode
2023/02/21 20:05:22 Exploring scheme for database "NOAA_water_database"
2023/02/21 20:05:22 fetching fields: command: "show field keys"; database: "NOAA_water_database"; retention: "autogen"
2023/02/21 20:05:22 found 5 fields; skipped 2 non-numeric fields
2023/02/21 20:05:22 fetching series: command: "show series"; database: "NOAA_water_database"; retention: "autogen"
2023/02/21 20:05:22 found 14 series
2023/02/21 20:05:22 explore query failed: can't find field keys for measurement "h2o_feet"
```
This creates a new database, sets up a retention policy, copys one measurement to a temporary, deletes all data in the measurement and copys the data back into the retention policy.
In the end you have a retention policy autogen with no data in the h2o_feet measurement and at this moment vmctl struggels to import the data.

**Comment by dmitryk-dk** [comment]
Thank you! Will take a look how to change the logic to fix the issue with measurement

**Comment by madface303** [comment]
Hi @dmitryk-dk, please excuse my poor knowledge in github, but i have no idea how to build this.
Do you have a short guidiance or can give me a link to a downloadable file?
My actual setup is a raspberry pi 4, so i would need it for arm 32 bit.
Thanks in advance!

**Comment by dmitryk-dk** [comment]
Hi @madface303 ! 
I can do both :
1. Clone VM repository to your local machine 
2. Checkout to this commit
```bash
git checkout 8acc15c75ed0c816b84b49d84f6371bb3c04dafb
```
You should see
```
HEAD is now at 8acc15c75 app/vmctl: skip series if measurement not found
```
3. From the root of the repository on your local machine, just run 
```bash
make vmctl-linux-arm
```
This command will produce a binary file and save it into `bin` folder of the repository
4. Go to the `bin` folder, and here you can find `vmctl-linux-arm` file
You can use it

I can send this file to your email

**Comment by madface303** [comment]
Hi @dmitryk-dk Thank you very much. I also had to install go in the actual version to get rid of some errors while making, but in the end i made it and had a working bin file.
The changes you made works for my case, it shows the errors but continues and it looks like this:
```
./vmctl-linux-arm influx --influx-database homeassistant --vm-addr http://localhost:8428
InfluxDB import mode
2023/02/26 13:03:52 Exploring scheme for database "homeassistant"
2023/02/26 13:03:52 fetching fields: command: "show field keys"; database: "homeassistant"; retention: "autogen"
2023/02/26 13:03:52 found 62 fields; skipped 89 non-numeric fields
2023/02/26 13:03:52 fetching series: command: "show series"; database: "homeassistant"; retention: "autogen"
2023/02/26 13:03:52 found 405 series
2023/02/26 13:03:52 can't find field keys for measurement " "
2023/02/26 13:03:52 can't find field keys for measurement "BTCEUR"
2023/02/26 13:03:52 can't find field keys for measurement "ETHEUR"
2023/02/26 13:03:52 can't find field keys for measurement "ETHUSD"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
2023/02/26 13:03:52 can't find field keys for measurement "bytes"
Found 2012 timeseries to import. Continue? [Y/n]
VM worker 0:↓ 16278 samples/s
VM worker 1:↓ 13597 samples/s
Processing series: 2012 / 2012 [███████████████████████████████████████████████████████████████████████████████] 100.00%
2023/02/26 13:07:07 Import finished!
2023/02/26 13:07:07 VictoriaMetrics importer stats:
  idle duration: 5m11.116606244s;
  time spent while importing: 3m13.848641012s;
  total samples: 8558676;
  samples/s: 44151.33;
  total bytes: 180.5 MB;
VM worker 0:↙ 15228 samples/s
VM worker 1:↙ 12719 samples/s
Processing series: 2012 / 2012 [███████████████████████████████████████████████████████████████████████████████] 100.00%
2023/02/26 13:07:07 Total time: 3m14.793065692s
```
I double checked the import ran fine by setting up an instance of vm on the raspberry, the data seems to be there when i query, so i think you can merge the PR.
Many many thanks for the help and the really fast responses, for me the issue is solved and it could be closed.

**Comment by dmitryk-dk** [comment]
Thank you for your test! It will be closed when PR is approved and merged.

**Comment by valyala** [comment]
The bugfix has been included in [vmctl v1.88.1](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed.

**Comment by valyala** [comment]
The bugfix has been also included in [v1.78.3 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.3).
